en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authorization/opa-validation.md

```diff
@@ -0,0 +1,203 @@
+
+| Numbers | Description                                                                                                                                         |
+|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1       | Client request                                                                                                                                      |
+| 2       | Request to validate (i.e. authentication, rate-limiting, OPA validation and other validations) the client request through Enforcer                  |
+| 3       | Enforcer calling the OPA server with the JSON payload described in [Request Payload to the OPA server](#request-payload-to-the-opa-server)          |
+| 4       | Response from OPA server after validating the request as described in [Response Payload from the OPA server](#response-payload-from-the-opa-server) |
+| 5       | Respond the validation status to the Router                                                                                                         | 
+| 6,7     | Response from the backend                                                                                                                           |
+| 8       | Response to the client                                                                                                                              |
+
+## Attaching OPA Policy to an API Resource
+
+By attaching an OPA policy to the API Resource you can enforce OPA validation to that API resource with providing the following values in the policy.
+
+| Name              | Description                                                   | Sample                          |
+|-------------------|---------------------------------------------------------------|---------------------------------|
+| Server URL        | URL of the OPA Server                                         | `http://localhost:8181/v1/data` |
+| Policy            | Name of the policy                                            | `abc-policy`                    |
+| Rule              | Name of the rule                                              | `xyz-rule`                      |
+| Request Generator | **Optional** fully qualified name of the implementation class | Empty string for Default class  |
+| Access Token      | **Optional** access token for OPA server authentication       | `my-secret-token`               |
+
+<!-- // TODO: (renuka) link configuring OPA from APIM (still not in 410-alpha) when APIM docs available -->
+
+## Defining Policies in the OPA Server
+
+You can define you own policy enforcement logic in OPA by using the values provided by the Choreo Connect Enforcer. Following is a sample policy definition in **Rego** to check the header value `x-abcd` should be `ABCD` when HTTP method is `PUT`, otherwise HTTP method can be `POST`.
```

Comment:
```suggestion
You can define your own policy enforcement logic in OPA by using the values provided by the Choreo Connect Enforcer. The following is a sample policy definition in **Rego** to check the header value. `x-abcd` should be `ABCD` when HTTP method is `PUT`, otherwise HTTP method can be `POST`.
```